File: ClientApp/app/models/fighter.ts

export interface skills {
    id:number,
    name:string,
    description:string,
    attackrange:number

}

export interface Fighter {
    id:number,
    name:string,
    power:number,
    speed:number,
    skills:skills[],
    isFinalForm:boolean,
    DateOfBirth:string
}

export interface SaveFighter {
  id: number;
  name: string;
  power: number;
  speed: number;
  skills: number[];
  isFinalForm: boolean;
  DateOfBirth: string;
}

File: Core/Models/Fighter.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;

namespace Vega.Core.Models
{
    public class Fighter
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public int Power { get; set; }
        [Required]
        public int Speed { get; set; }
        [Required]
        public bool IsFinalForm { get; set; }
        [Required]
        public DateTime DateOfBirth { get; set; }
        public ICollection<FighterSkill> Skills { get; set; }
        public Fighter()
        {
            Skills  = new Collection<FighterSkill>();
        }


    }
}